app/scripts/background.js

```diff
@@ -268,7 +268,24 @@ async function initialize() {
     await DesktopManager.init(platform.getVersion());
     ///: END:ONLY_INCLUDE_IN
 
-    setupController(initState, initLangCode);
+    let { isFirstMetaMaskControllerSetup } = await browser.storage.session.get([
+      'isFirstMetaMaskControllerSetup',
+    ]);
+
+    if (isFirstMetaMaskControllerSetup === true) {
+      isFirstMetaMaskControllerSetup = false;
+      await browser.storage.session.set({ isFirstMetaMaskControllerSetup });
```

Comment:
Optional: we can isolate the setter out since it will be set for both conditions

Reply:
@DDDDDanica do you mean something like this?

```
if (isFirstMetaMaskControllerSetup === true) {
  await browser.storage.session.set({ isFirstMetaMaskControllerSetup });
} else if (isFirstMetaMaskControllerSetup === undefined) {
  await browser.storage.session.set({ isFirstMetaMaskControllerSetup });
}
isFirstMetaMaskControllerSetup = isFirstMetaMaskControllerSetup === undefined
	? true
	: isFirstMetaMaskControllerSetup ? false : true;
```

Reply:
Actually it would be like 

```suggestion
   if (isFirstMetaMaskControllerSetup === true) {
      isFirstMetaMaskControllerSetup = false;
   } else if (isFirstMetaMaskControllerSetup === undefined) {
      isFirstMetaMaskControllerSetup = true;
    }
      await browser.storage.session.set({ isFirstMetaMaskControllerSetup });
```

Reply:
Actually it would be like 

```suggestion
     isFirstMetaMaskControllerSetup = isFirstMetaMaskControllerSetup ? false : true;
      await browser.storage.session.set({ isFirstMetaMaskControllerSetup });
```

Reply:
okay, I understand what you are saying now

So what we want is:
- if `isFirstMetaMaskControllerSetup` is `false`, leave it `false`
- if it is `true`, set it to `false`
- if it is `undefined`, set it to `true`

so we would at least have to have this part:
```
isFirstMetaMaskControllerSetup = isFirstMetaMaskControllerSetup === undefined
	? true
	: isFirstMetaMaskControllerSetup ? false : true;
```

Reply:
but I think the current way it is written might be clearer... I can simplify it a little bit though

Reply:
I agree, the original statement is more clear. But what i think can be extracted is this line:
`await browser.storage.session.set({ isFirstMetaMaskControllerSetup });`

Reply:
oh, actually, this can be way simpler

I think it can reduced to just

```
    await browser.storage.session.set({
      isFirstMetaMaskControllerSetup: isFirstMetaMaskControllerSetup === undefined
    });
```